### PR TITLE
Add Next UI and Fixes to pre-commit

### PR DESCRIPTION
**Pull request by afkcodes**
## Description

This PR adds NEXT UI support, also fixes minor issues with Pre-commit not being run.  I have updated the eslint file as well to ensure some best practices.

Fixes #(Insert Issue Here Without Brackets)

## Changes Include

- [x] Bug fix 🐛 (non-breaking change which fixes an issue)
- [x] New feature ✨ (non-breaking change which adds functionality)
- [x] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)
- [x] Tooling ⚙️ (Changes to the build process or auxiliary tools and libraries such as documentation generation)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] ran and tested

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: .eslintrc.js
Comment (Jarmos-san) [inline .eslintrc.js:1]: Let's not increase the complexity of the project right now & stick to what Next.js provides by default. So, could you revert back to the older `.eslintrc.json`? We'll review the configurations after the first iteration of the landing page is complete.

**Review by Jarmos-san — CHANGES_REQUESTED** [review]

File: src/pages/index.tsx
Comment (Jarmos-san) [reply src/pages/index.tsx:6]: I would also like to add, there's an merge conflict on `package.json` blocking the PR. Since, PostCSS is a TailwindCSS dependency, removing that will also unblock the PR. So, if we can reach an consensus here then #123 can be closed.

File: .eslintrc.js
Comment (afkcodes) [reply .eslintrc.js:1]: code will be messy, and i don't want that.

File: .husky/pre-commit
Comment (afkcodes) [reply .husky/pre-commit:4]: will add

File: package.json
Comment (afkcodes) [reply package.json:12]: one works with yarn and one works with npm

File: src/pages/index.tsx
Comment (afkcodes) [reply src/pages/index.tsx:6]: Tailwind CSS and Next UI are two different things, we can remove either. tailwind is a CSS library that gives you classes to work and make things easy so that you don't have to write all your CSS.

Next UI is a component library it gives you the components out of the box that would be symmetric in the whole project.
There is no complexity being added here, IMO, these are like basic things that we do daily in frontent.

File: .prettierrc
Comment (afkcodes) [reply .prettierrc:2]: all these configs has been taken from the best opensource projects, i would suggest to checkout this branch see all things working first-hand.

Comment (Jarmos-san) [reply .prettierrc:2]: > ...all these configs have been taken from the best opensource projects...

No doubt there but let's focus on developing the website's source code first rather than worry about configuring Prettier :laughing:. Relying on the [standard defaults Prettier provides](https://prettier.io/docs/en/options.html) is good enough for now.

File: package.json
Comment (Jarmos-san) [reply package.json:12]: Is the added complexity a necessity? If we support Yarn as well then we'll need to make documentation changes as well which I think shouldn't be priority right now lol. I would really like to get started developing the website ASAP instead of tinkering around with the trivial stuffs.

So, let's stick to `npm` stuff for now.

File: src/pages/index.tsx
Comment (Jarmos-san) [reply src/pages/index.tsx:6]: Yup, we can remove TailwindCSS then in favour of NextUI.

I just have one little question though. The classes, `text-3xl`, `font-bold` & so on, are they TailwindCSS classes or from NextUI?

Comment (afkcodes) [reply src/pages/index.tsx:6]: those classes are from tailwind and the button component from Next UI

Comment (Jarmos-san) [reply src/pages/index.tsx:6]: We'll be removing TailwindCSS in favour of NextUI just so you know. Since the `Button` component uses TailwindCSS classes the website will break.

File: package.json
Comment (Jarmos-san) [reply package.json:12]: I doubt we need to support `yarn` since we specifically mentioned in the docs to use `npm` instead. So, could you make those changes as well?

File: src/pages/index.tsx
Comment (afkcodes) [reply src/pages/index.tsx:6]: then how will you style components tell me

File: package.json
Comment (afkcodes) [reply package.json]: you cant force any dev/contributors to use npm specifically, this repo is becoming opiniated.

File: src/pages/index.tsx
Comment (afkcodes) [reply src/pages/index.tsx:6]: Next UI is a component library, you can style them using basic CSS, but that is a hectic task.
Tailwind CSS gives you classes that will eventually convert to vanilla CSS.

Comment (Manav-SM) [reply src/pages/index.tsx:6]: We will need both Next UI and Tailwind CSS @Jarmos-san

Comment (Jarmos-san) [reply src/pages/index.tsx:6]: > We will need both Next UI and Tailwind CSS @Jarmos-san

Cool, let's keep it open & review this decision for later. But I don't over-complicating the task at hand is the way to go when we don't even have an MVP in place.

**Comment by Manav-SM** [comment]
We need to finalize the stack we will use before doing any more commits.

**Pull request closed without merging**
